### PR TITLE
School homepage + pupil edit changes

### PR DESCRIPTION
**Pull request by pris54**

### CODE REVIEW

File: admin/views/school/_pupil-form.ejs [hunk 1]
```diff
@@ -88,6 +87,14 @@ if (formData._id) {
                        name="upn"
                        value="<%= formData['upn'] %>"
                 />
+                <div class="upn-text">
```
Comment (jon-shipley) [inline admin/views/school/_pupil-form.ejs]: Please revert this change - I have a different implementation that has been approved by UX.  This should be using <detail><summary> tags.

File: admin/assets/stylesheets/elements/_pupils.scss [hunk 2]
```diff
@@ -33,3 +33,32 @@
     width: 60%;
   }
 }
+
+/* Add/edit form */
+
+.upn-text {
```
Comment (jon-shipley) [inline admin/assets/stylesheets/elements/_pupils.scss]: Please revert the style change associated with the changes to the UPN text on add/edit pupil.  This is covered under a different ticket using `<detail><summary>` tags.

File: admin/assets/javascripts/global-scripts.js [hunk 3]
```diff
@@ -30,7 +30,13 @@ $(function () {
   if ($('#pupilsList').length > 0) disableCheckAll('#pupilsList')
   if ($('#attendanceList').length > 0) disableCheckAll('#attendanceList')
 
-  $('input:file').on('change', function () {
+  $('.upn-text > a').on('click', function () {
```
Comment (jon-shipley) [inline admin/assets/javascripts/global-scripts.js]: Please revert the style change associated with the changes to the UPN text on add/edit pupil. This is covered under a different ticket using `<detail><summary>` tags

File: admin/package.json [hunk 4]
```diff
@@ -8,14 +8,16 @@
     "start": "node ./server.js",
     "postinstall": "gulp build",
     "test": "jasmine-node spec --verbose --color --captureExceptions --forceexit",
-    "coverage": "nyc jasmine-node spec --verbose --color --captureExceptions --forceexit"
+    "coverage": "nyc jasmine-node spec --verbose --color --captureExceptions --forceexit",
+    "startn": "nodemon ./server.js",
+    "watch-jscss": "gulp watch"
```
Comment (jon-shipley) [inline admin/package.json]: Good 👍 

But why is this called `watch-jscss`? What does the `j` mean?

File: admin/views/school/_pupil-form.ejs [hunk 5]
```diff
@@ -49,7 +48,7 @@ if (formData._id) {
             </div>
 
             <div class="form-group <%= err.isError('middleNames') ? 'form-group-error' : '' %>">
-                <label class="form-label-bold" for="middleNames">Middle name(s)
+                <label class="form-label-bold" for="middleNames">Middle name(s) <span class="font-xsmall">(optional)</span>
```
Comment (jon-shipley) [inline admin/views/school/_pupil-form.ejs]: Please revert.  Handled on another ticket.

File: admin/views/school/_pupil-form.ejs [hunk 6]
```diff
@@ -177,8 +184,13 @@ if (formData._id) {
...
             <div class="form-buttons">
                 <input type="submit" class="button" value="<%= submitLabel %>"/>
-                <a href="/school/manage-pupils" class="button button-secondary">Don’t save</a>
+                <a href="/school/manage-pupils" class="button button-secondary">Cancel</a>
             </div>
         </form>
+
+        <br />
+        <p>
+            <a href="#top">Back to top</a>
```
Comment (jon-shipley) [inline admin/views/school/_pupil-form.ejs]: Good 👍

**Comment by jon-shipley** [comment]
@paula - please can you also update the yarn lock file?  This is required by the travis build.  Travis has 1st class support for yarn.

File: admin/package.json (hunk 4 above)
Comment (pris54) [reply admin/package.json]: Most likely a typo

Comment (jon-shipley) [reply admin/package.json]: Do you want to remove the `j` then?  Personally I would go with `sasswatch` as this is the name in the pupil-spa package.json

**Comment by jon-shipley** [comment]
The school home controller unit tests are missing.

**Comment by pris54** [comment]
@jon-shipley watch-jscss edited to sasswatch (in the README.md file too).

**Comment by pris54** [comment]
NOTE RE' FAILING TEST: Mohsenq is looking into the broken tests.

**Comment by pris54** [comment]
@jon-shipley screenshots removed

**Comment by jon-shipley** [comment]
@pris54 - please can you also remove the test report: `pupil/test/reports/report.html`?
